Refactor file structure

lishaduck opened the pull request:
Also, fixes current_directory/cwd error, and rewrites the readme to use markdown, @infinityJKA. Reformats the code with black, sorts imports with isort, and manages dependencies with poetry. Adds a gitignore. Adds status checks.

lishaduck committed:
remove gitignored files

lishaduck committed:
add gitignore

lishaduck committed:
delete data files

lishaduck committed:
feat(github): add dependabot.yml

lishaduck committed:
feat(poetry): initialize poetry

lishaduck committed:
switch readme.txt -> readme.md

lishaduck committed:
move game to bipolev folder

lishaduck committed:
add black and isort

lishaduck committed:
reformat files

lishaduck committed:
add docstring to init

lishaduck committed:
fix: imports

lishaduck committed:
fix: add some typing to remove unsafe defaults

lishaduck committed:
fix: add encoding to 'open's

lishaduck committed:
fix: current version in pyproject.toml

lishaduck committed:
feat(ci/cd): create a release workflow

lishaduck committed:
update gitignore

lishaduck committed:
refactor: bipoleV/ -> bipole5/

lishaduck committed:
format the toml

lishaduck committed:
fix: current_directory, version on start page

lishaduck committed:
feat: add back saving

lishaduck committed:
fix: fix the gitignore again

lishaduck committed:
docs: update readme

lishaduck commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,11 @@
+
+version: 2
+updates:
+  - package-ecosystem: "pip" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "daily"
```
Checks for dep updates daily

lishaduck commented on .gitignore:
```diff
@@ -0,0 +1,312 @@
+# THESE SHOULD BE IGNORED, BUT ARE NOT YET, DUE TO LACK OF PROPER GENERATION AT STARTUP
+#save.dat
+#autosave.dat
```
```suggestion
save.dat
autosave.dat
```

lishaduck commented on README.md:
```diff
@@ -0,0 +1,36 @@
+
+! ! ! CAPS LOCK WILL MESS WITH INPUTS, DON'T HAVE CAPS LOCK ENABLED ! ! !
+
+| Key        | Mapping         |
+| ---------- | --------------- |
+| 1          | SAVE            |
+| 2          | TOGGLE SIDESTEP |
+| 3          | STAT            |
+| 4          | KEY ITEMS       |
+| q          | A               |
+| e          | B               |
+| w          | UP              |
+| r          | EQUIP           |
+| t          | PARTY           |
+| a          | LEFT            |
+| s          | DOWN            |
+| d          | RIGHT           |
+| f          | ITEM            |
+| g          | TALK            |
+| arrow keys | movement        |
+| space key  | for TALK        |
```
Converted the list to a table

lishaduck replied:
Converted the list to a table.

lishaduck commented on bipole5/__init__.py:
```diff
@@ -0,0 +1,4 @@
+"""Bipole V! A game.
+
+TODO: Ought BipoleV.py to be renamed to __init__.py, and replace this?
```
?

lishaduck commented on pyproject.toml:
```diff
@@ -0,0 +1,33 @@
+version = "1.0.0"
+
+[tool.poetry.dependencies]
+numpy = "^1.23.5"
+pillow = "^9.3.0"
+python = "^3.10"
```
Dependencies for the project.

lishaduck replied:
```suggestion
numpy = "^1.23.5"
pillow = "^9.3.0"
python = "^3.7"
```

lishaduck committed:
Update minimum python version

lishaduck commented on pyproject.toml:
```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
+authors = ["InfinityJKA <jkajustin@gmail.com>"]
```
Is this the right email?

lishaduck commented on bipole5/BipoleV.py:
```diff
@@ -1,95 +1,165 @@
+
+from PIL import Image, ImageTk
+
+current_directory: Path = Path(os.path.realpath(__file__)).parent
```
This is a better way to get the file's location, @infinityJKA.

lishaduck commented on bipole5/BipoleV.py:
```diff
@@ -749,29 +1310,39 @@ def disable_keys():
-    dialogue = [line.rstrip('\n') for line in file_contents]
+    file_contents = open(
+        current_directory
+        / "dialogue"
+        / maps.current_location[4]
+        / (input_file + ".txt"),
```
Use the '/' operator to append strings to a path, @infinityJKA.

lishaduck commented on bipole5/BipoleV.py:
```diff
@@ -749,29 +1310,39 @@ def disable_keys():
     dialogue_index = 1
-    file_contents = open(current_directory+"/dialogue/"+maps.current_location[4]+"/"+input_file+".txt")
-    dialogue = [line.rstrip('\n') for line in file_contents]
+    file_contents = open(
+        current_directory
+        / "dialogue"
+        / maps.current_location[4]
+        / (input_file + ".txt"),
+        encoding="UTF-8",
+    )
```
Use the '/' operator to append strings to a path, @infinityJKA.
Oh, and when opening in text ("t") mode, always specify an encoding.

lishaduck commented:
What did you use to build the output folder, @infinityJKA?

infinityJKA commented:
I think I used auto-py-to-exe

On Mon, Dec 19, 2022 at 10:11 AM Eli ***@***.***> wrote:

> What did you use to build the output folder, @infinityJKA
> <https://github.com/infinityJKA>?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/infinityJKA/bipoleV/pull/1#issuecomment-1357900726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALNZ2I5JRV47VOQ664E7OFLWOCCLXANCNFSM6AAAAAATA6BOK4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

lishaduck committed:
add auto-py-to-exe

I tried it, and it seemed to work.
However, I'm on a managed mac, so I'm not sure.
It wouldn't let me run it.

lishaduck committed:
update publish.yml

lishaduck commented:
I think packaging works now, @infinityJKA. The only thing would be uploading the build artifacts to releases. I'd think there's a way to automate it, but I don't see one.

lishaduck committed:
Update publish.yml

lishaduck committed:
feat: use pyinstaller, not auto-py-to-exe

auto-py-to-exe is a thin *graphical* wrapper over pyinstaller.
However, we aren't we aren't using npm and puppeteer, so we need to use pyinstaller ourselves.

lishaduck commented:
Packaging should be fixed now, @infinityJKA.

lishaduck committed:
fix: move save files

lishaduck committed:
fix: remove deprecated image tools

lishaduck commented:
Should I break this into smaller PRs, 1 at a time, @infinityJKA?

infinityJKA commented:
> Should I break this into smaller PRs, 1 at a time, @infinityJKA?

Idk, if you want

lishaduck commented:
I've decided that I'm too busy to update this, and it's useless in this condition, so I'm closing it.

infinityJKA commented:
real

lishaduck commented:
I'm going to prune the repo at the very least now that I can run it again. I'll try to be more conservative this time.

infinityJKA commented:
@lishaduck should i be scared

lishaduck commented:
No. I'm going 1 commit @ a time

infinityJKA commented:
Remember to commit responsibly!

lishaduck commented:
I've double checked, #2 still works fully on a managed mac w/py3.11.6!

lishaduck commented:
#3 is just docs, but I checked anyway, and it too works.
